src/underactuated/meshcat_visualizer.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf8 -*-
+
+'''
+
+Usage:
+
+MeshcatRigidBodyVisualizer acts identically to PyPlotVisualizer
+from the perspective of the code: it is a System block that consumes
+system state and visualizes that state at every publish step.
+
+However, where PyPlotVisualizer opens its own visualizer window,
+MeshcatRigidBodyVisualizer expects the command `meshcat-server`
+to already be running in another terminal. The visualization
+will be available at the web url provided by that server in real
+time (either when the system is being simulated, or when the
+animate() method is called).
+
+'''
+
+import argparse
+import math
+import os.path
+import time
+
+import numpy as np
+
+from pydrake.all import (
+    Context,
+    DiagramBuilder,
+    FindResourceOrThrow,
+    LeafSystem,
+    PortDataType,
+    Quaternion,
+    RigidTransform,
+    RotationMatrix,
+    SignalLogger,
+    Simulator,
+)
+
+from pydrake.common import FindResourceOrThrow
+from pydrake.geometry import (
+    GeometryVisualizationImpl, SceneGraph)
+from pydrake.lcm import DrakeLcm
+from pydrake.multibody.multibody_tree import UniformGravityFieldElement
+from pydrake.multibody.multibody_tree.multibody_plant import MultibodyPlant
+from pydrake.multibody.multibody_tree.parsing import AddModelFromSdfFile
+from pydrake.systems.framework import DiagramBuilder
+from pydrake.systems.analysis import Simulator
+
+from underactuated.utils import FindResource
+
+import meshcat
+import meshcat.transformations as tf
+
+
+class MeshcatVisualizer(LeafSystem):
+    def __init__(self,
+                 scene_graph,
+                 draw_timestep=0.033333,
+                 prefix="SceneGraph",
+                 zmq_url="tcp://127.0.0.1:6000"):
+        LeafSystem.__init__(self)
+
+        load_robot_msg = GeometryVisualizationImpl.BuildLoadMessage(
+            scene_graph)
+
+        self.set_name('meshcat_visualization')
+        self.timestep = draw_timestep
+        self._DeclarePeriodicPublish(draw_timestep, 0.0)
+
+        # Pose bundle (from SceneGraph) input port.
+        self._DeclareInputPort(PortDataType.kAbstractValued, 0)
+
+        # Set up meshcat.
+        self.prefix = prefix
+        self.vis = meshcat.Visualizer(zmq_url=zmq_url)
+        self.vis[self.prefix].delete()
+
+        for i in range(load_robot_msg.num_links):
+            link = load_robot_msg.link[i]
+            [source_name, frame_name] = link.name.split("::")
+
+            for j in range(link.num_geom):
+                geom = link.geom[j]
+                element_local_tf = RigidTransform(
+                    RotationMatrix(Quaternion(geom.quaternion())),
+                    geom.position()).GetAsMatrix4()
+
+                if geom.type == 1:  # geom.BOX:
```

Comment:
I finally punted after lots of failed attempts to bind the constexpr from the lcm type.  (pybind was choking on it, saying at runtime that the symbol was missing, presumably because the compiler had optimized it out?).  pybind's class_ doesn't have the value() methods that the enum type would have used to support it.  

i had one more idea i can try (binding a method BOX() instead), but will see what @EricCousineau-TRI is up to.

Reply:
I didn't appreciate that you're dealing with a pybind-bound C++ LCM message, yuck. Nevermind :(